### PR TITLE
plan: enforce property for index join.

### PR DESCRIPTION
**Pull request by hanfei1991**
fix bug
@shenli @coocood @zimulala @winoros PTAL

### CODE REVIEW

**Comment by winoros** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

File: executor/join_test.go [hunk 1]
```diff
@@ -437,6 +437,14 @@ func (s *testSuite) TestSubquery(c *C) {
 	result.Check(testkit.Rows("1 0", "2 1"))
 	result = tk.MustQuery("select (select count(*) from t k where t.id = id) from s, t where t.id = s.id limit 1")
 	result.Check(testkit.Rows("1"))
+	tk.MustExec("drop table if exists t")
```
Comment (ngaut) [inline executor/join_test.go]: Using sql snippet instead of call MustExec multiple times.

**Comment by hanfei1991** [comment]
@ngaut PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**